FogCreek/Glitch-Community
# chestnut-headphones: Add onboarding banner to homepage and profile page

## Links
* [chestnut-headphones.glitch.me/](https://chestnut-headphones.glitch.me/)
* [Onboarding design spec](https://docs.google.com/document/d/1iiJnU8COhu0EreV9i5xHaBMJRgeclaxLw6liBz-CNqU/)

![welcome](https://user-images.githubusercontent.com/7584833/63360502-7b900180-c33c-11e9-9024-2eca465489c5.gif)

## GIF/Screenshots:
<table>
<tr>
     <td><strong>large viewport</strong></td>
     <td><img src="https://user-images.githubusercontent.com/7584833/63358952-e12ebe80-c339-11e9-8105-05287cd31ca3.png"/>
  </td>
  </tr>
<tr>
     <td><strong>medium viewport</strong></td>
     <td><img src="https://user-images.githubusercontent.com/7584833/63358905-c8bea400-c339-11e9-8d9f-dbf2c531fbbd.png" />
  </td>
  </tr>
  <tr>
     <td><strong>small viewport</strong></td>
     <td><img height="550" src="https://user-images.githubusercontent.com/7584833/63358764-92812480-c339-11e9-96a6-6e778edc57e3.png"/></td>
  </tr>
</table>

## Changes:
* Add OnboardingBanner component
* Add CategoriesGrid component (used in OnboardingBanner and /create)
* Move shared constants from /create and the homepage to constants.styl
* Move NewProjectPop out of /header
* Update NewProjectPop to accept props for button styles and button text
* Update NewProjectPop analytics to send user id and the source of the clicks, i.e., navbar, onboarding banner, etc.

## How To Test:

Log in with a test account that doesn't have any projects
- You should see the onboarding banner on the homepage and on your profile page
- Change your screen width, the banner should look good
- Use the accessibility tools to inspect the welcome illustration. The label should be "Welcome"
- Click the "Create New Project" button, it should work.
- Click all the links in the onboarding banner, they should work.

Create a project
- You should not see the onboarding banner

Log out
- You should not see the onboarding banner

Various things to check
- The categories grid on /create still looks good
- The "New Project" button in the navbar still works

## Feedback I'm looking for:
- @ttseng does the banner look good to you?
- @sarahzinger does the data I'm sending along in the new project pop look good?

## Before deploying
- [ ] Run accessibility insights
